server/model/test_run_event.go

```diff
@@ -59,6 +129,60 @@ func (e TestRunEvent) ResourceID() string {
 	return fmt.Sprintf("test/%s/run/%d/event", e.TestID, e.RunID)
 }
 
+func (e TestRunEvent) WithDataStoreConnection(dataStoreConnection ConnectionResult) TestRunEvent {
+	e.DataStoreConnection = dataStoreConnection
+	return e
+}
+
+func (e TestRunEvent) WithPolling(pollingInfo PollingInfo) TestRunEvent {
+	e.Polling = pollingInfo
+	return e
+}
+
+func (e TestRunEvent) WithOutputs(outputs ...OutputInfo) TestRunEvent {
+	e.Outputs = outputs
+	return e
+}
+
+func NewTestRunEvent(stage TestRunEventStage, eventType TestRunEventType, testID id.ID, runID int) TestRunEvent {
+	return NewTestRunEventWithArgs(stage, eventType, testID, runID, map[string]string{})
+}
+
+func NewTestRunEventWithArgs(stage TestRunEventStage, eventType TestRunEventType, testID id.ID, runID int, titleAndDescriptionArgs map[string]string) TestRunEvent {
+	eventTypeTable, found := eventTable[stage]
```

Comment:
Is this for events that have extra information attached to them?

Reply:
Yes, by following @mathnogueira we have a less complex way to do that, for instance here: https://github.com/kubeshop/tracetest/pull/2280/files#diff-cc7adc54283b5ac608363703f8b26f6b0c084cf35c520e7d06ed20da9dd0fadeR206